### PR TITLE
Custom reporter

### PR DESCRIPTION
**Pull request by suryart**
This PR adds:

- Command to generate the build report: `generate-report <build-id>`
- Interface for fetching the report data in JSON from the API.
- Saving JSON and HTML file that's generated in the same directory where the command was run.
- Always generates the reports in sync CLI mode.

### CODE REVIEW

File: bin/commands/generateReport.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    let buildId = args._[1];
+
+    let options = {
+      url: `${config.buildUrl}${buildId}/custom_report`,
+      auth: {
+        user: bsConfig.auth.username,
+        password: bsConfig.auth.access_key,
+      },
+      rejectUnauthorized: false,
```
Comment (suryart) [inline bin/commands/generateReport.js:37]: Remove this once done with testing.

File: bin/helpers/reporterHTML.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        message = Constants.userMessages.API_DEPRECATED;
+        logger.info(message);
+      }
+    } else if (resp.statusCode != 200) {
+      messageType = Constants.messageTypes.ERROR;
+      errorCode = 'api_failed_build_generate_report';
+
+      if (build) {
+        message = `${
+          Constants.userMessages.BUILD_GENERATE_REPORT_FAILED.replace('<build-id>>', buildId)
```
Comment (SagarGaniga) [inline bin/helpers/reporterHTML.js:140]: `<build-id>>` should this be `<build-id>`

Comment (suryart) [reply bin/helpers/reporterHTML.js]: Done 👍

**Review by SagarGaniga — APPROVED** [review]

**Comment by prateekbstack** [comment]
LGTM

**Pull request merged**